modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/JavaUpgradeFetchCPDefinitionByCProductExternalReferenceCodeCheck.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.petra.string.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Michael Cavalcanti
+ */
+public class JavaUpgradeFetchCPDefinitionByCProductExternalReferenceCodeCheck
+	extends BaseJavaTermCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, JavaTerm javaTerm,
+			String fileContent)
+		throws Exception {
+
+		List<String> importNames = javaTerm.getImportNames();
+
+		String content = javaTerm.getContent();
+
+		if (!importNames.contains(
+				"com.liferay.commerce.product.model.CPDefinition")) {
+
+			return content;
+		}
+
+		Matcher methodCallMatcher = _methodCallPattern.matcher(content);
+
+		while (methodCallMatcher.find()) {
+			String methodCall = JavaSourceUtil.getMethodCall(
+				content, methodCallMatcher.start());
+
+			if (_checkMethodCall(content, fileContent, methodCall)) {
+				content = StringUtil.replace(
+					content, methodCallMatcher.group(),
+					_reorderParameters(methodCall, methodCallMatcher.group(1)));
+			}
+		}
+
+		return content;
+	}
+
+	@Override
+	protected String[] getCheckableJavaTermNames() {
+		return new String[] {JAVA_METHOD};
+	}
+
+	private boolean _checkMethodCall(
+		String content, String fileContent, String methodCall) {
+
+		List<String> parameterList = JavaSourceUtil.getParameterList(
+			methodCall);
```

Comment:
Just in case, I think we need to check the number of arguments here and ensure we are dealing with 2 arguments. We are making the assumption that the method call has at least 2 or more arguments.